bson/src/test/unit/org/bson/vector/VectorGenericBsonTest.java

```diff
@@ -0,0 +1,272 @@
+                actualVector);
+        assertEquals(expectedPadding, actualVector.getPadding());
+    }
+
+    private void assertVectorDecoding(final float[] expectedVectorData,
+                                      final Vector.DataType expectedDType,
+                                      final Float32Vector actualVector) {
+        float[] actualVectorArray = actualVector.getVectorArray();
+        Assertions.assertArrayEquals(actualVectorArray, expectedVectorData,
+                () -> "Actual: " + Arrays.toString(actualVectorArray) + " !=  Expected:" + Arrays.toString(expectedVectorData));
+        assertEquals(expectedDType, actualVector.getDataType());
+    }
+
+    private byte[] toByteArray(final BsonArray arrayVector) {
+        byte[] bytes = new byte[arrayVector.size()];
+        for (int i = 0; i < arrayVector.size(); i++) {
+            bytes[i] = (byte) arrayVector.get(i).asInt32().getValue();
+        }
+        return bytes;
+    }
+
+    private float[] toFloatArray(final BsonArray arrayVector) {
+        float[] floats = new float[arrayVector.size()];
+        for (int i = 0; i < arrayVector.size(); i++) {
+            BsonValue bsonValue = arrayVector.get(i);
+            if (bsonValue.isString()) {
+                floats[i] = parseFloat(bsonValue.asString());
+            } else {
+                floats[i] = (float) arrayVector.get(i).asDouble().getValue();
+            }
+        }
+        return floats;
+    }
+
+    private static float parseFloat(final BsonString bsonValue) {
+        String floatValue = bsonValue.getValue();
+        switch (floatValue) {
+            case "-inf":
+                return Float.NEGATIVE_INFINITY;
+            case "inf":
+                return Float.POSITIVE_INFINITY;
+            default:
+                return Float.parseFloat(floatValue);
+        }
+    }
+
+    private static Stream<Arguments> provideTestCases() throws URISyntaxException, IOException {
+        List<Arguments> data = new ArrayList<>();
+        for (File file : JsonPoweredTestHelper.getTestFiles("/bson-binary-vector")) {
+            BsonDocument testDocument = JsonPoweredTestHelper.getTestDocument(file);
+            for (BsonValue curValue : testDocument.getArray("tests", new BsonArray())) {
+                BsonDocument testCaseDocument = curValue.asDocument();
+                data.add(Arguments.of(createTestCaseDescription(testDocument, testCaseDocument), testDocument, testCaseDocument));
+            }
+        }
+        return data.stream();
+    }
+
+    private static String createTestCaseDescription(final BsonDocument testDocument,
+                                                    final BsonDocument testCaseDocument) {
+        boolean isValidTestCase = testCaseDocument.getBoolean("valid").getValue();
+        String testSuiteDescription = testDocument.getString("description").getValue();
+        String testCaseDescription = testCaseDocument.getString("description").getValue();
```

Comment:
We usually call these `fileDescription` and `testDescription`. Let's use the same names here, especially given that https://github.com/mongodb/specifications/tree/master/source/bson-binary-vector/tests does not mention "suite".